extension/lib/shopgate/errorHandler.js

```diff
@@ -0,0 +1,32 @@
+const shopifyStorefrontErrorLogger = require('./pipeline/shopifyStorefrontErrorLogger')
+const ShopgatePipelineCallFailedError = require('./pipeline/CallFailedError')
+const {isAccessForbidden} = require('../shopify/StorefrontClient')
+const ShopgateAuthorisation = require('./Authorisation')
+
+/**
+ * @param {Error} err
+ * @param {PipelineContext} context
+ */
+module.exports = function (err, context) {
+  if (!shopifyStorefrontErrorLogger(err, context)) {
+    context.log.error({
+      msg: err.message,
+      stack: err.stack
+    })
+  }
+
+  if (isAccessForbidden(err)) {
+    ShopgateAuthorisation.create(context)
```

Comment:
Do you maybe want to await the release here or in method: releaseShopifyStorefrontToken before throwing an exception?

Reply:
I found this to be the easiest solution at this time in order to log an error in case it happens during release of the stored token. 
Having another try/catch in every pipeline implementation is another solution but seems a bit off as this would be then "errorHandler" throwing additional errors.